service/src/main/java/bio/terra/cbas/dependencies/leonardo/LeonardoClient.java

```diff
@@ -26,4 +26,10 @@ public ApiClient getApiClient() throws AzureAccessTokenException {
     apiClient.setDebugging(leonardoServerConfiguration.debugApiLogging());
     return apiClient;
   }
+
+  public ApiClient getApiClient(String accessToken) throws AzureAccessTokenException {
```

Comment:
Do we still need the `AzureAccessTokenException` if we're always using the user's bearer token?

Reply:
We do if we are still using the above `getAptClient()`as it still utilizes the AzureToken in the default header. As part of this PR, should we remove the default header here and utilize `.setAccessToken()` instead?

Reply:
My instinct would be yes since CBAS shouldn't need to be aware of an AzureToken any longer... probably good for someone else to confirm

Reply:
When I tried `setAccessToken` it didn't work or didn't seem to always apply it - hence switching to the default header. If you've double checked it's definitely getting added properly then 👍